app/components/candidate_interface/previous_applications_component.rb

```diff
@@ -0,0 +1,73 @@
+module CandidateInterface
+  class PreviousApplicationsComponent < ViewComponent::Base
+    include ViewHelper
+
+    def initialize(application_form:)
+      @application_form = application_form
+    end
+
+    def render?
+      application_choices.present?
+    end
+
+    def application_choices
+      (previous_application_choices + eligible_current_application_choices)
+        .compact
+        .sort_by(&:id)
+    end
+
+    def provider_name_for(application_choice)
+      application_choice
+        .course
+        .provider
+        .name
+    end
+
+    def course_name_and_code_for(application_choice)
+      name = application_choice.course.name
+      code = application_choice.course.code
+
+      "#{name} (#{code})"
+    end
+
+  private
+
+    def eligible_current_application_choices
+      if application_choice_has_accepted_state_present?
+        # Reject all applications that have an ACCEPTED_STATE
+        # These will appear in the CandidateInterface::CourseChoicesReview component
+        application_choices_without_accepted_states
+      else
+        []
+      end
+    end
+
+    def application_choices_without_accepted_states
+      current_application_choices
+        .reject { |ac| ApplicationStateChange::ACCEPTED_STATES.include?(ac.status.to_sym) }
+    end
+
+    def application_choice_has_accepted_state_present?
+      current_application_choices.any? { |ac| ApplicationStateChange::ACCEPTED_STATES.include?(ac.status.to_sym) }
+    end
+
+    def current_application_choices
+      @application_choices ||= @application_form
+        .application_choices
+        .includes(:course, :site, :provider, :offered_course_option, :interviews)
+        .order(id: :asc)
+    end
+
+    def previous_application_choices
+      previous_application_form = @application_form.previous_application_form
```

Comment:
The candidate could have multiple previous application_forms so I think this probably needs to be something like

```
candidate =  @application_form.candidate
previous_application_forms = candidate.application_forms - [candidate.current_application]

if previous_application_forms
    previous application_forms.map(&:application_choices).flatten.includes(course: provider)
end 

```

Reply:
Ah ok. I'll make the changes - thanks